models/recipe.js

```diff
@@ -0,0 +1,43 @@
+const mongoose = require("mongoose");
+
+const recipeSchema = mongoose.Schema(
+  {
+    title: {
+      type: String,
+      required: true,
+    },
+    picturePath: {
+      type: String,
+      default: "",
+    },
+    servings: {type: Number, required: true},
+    steps: [
+      {
+        stepName: {type: String},
+        step: [{type: String}],
+      },
+    ],
+    ingredients: [
+      {
+        ingredient: {type: String, required: true},
+        unit: {type: String},
+        amount: {type: Number, required: true},
+        section: {type: String},
+      },
+    ],
+    time: {hours: {type: Number}, minutes: {type: Number}},
+    userId: {
+      type: String,
+    },
+    isRecommended: {type: Map, of: Boolean},
```

Comment:
Why is this a map boolean?

Reply:
I heard this is a more efficient way of getting who liked it. Instead of looping all the users and finding out which user liked the recipe, you just record the boolean, and count that? Kinda forgot, need to double-check. I think  O(n) vs O(1) ?

Reply:
so the key is the user?

Reply:
I think so? So it needs to check if the user is in that map. If the user likes it, it will be added to the map, if they dont like it, will remove the map? o.o